src/data/int/cast.lean

```diff
@@ -198,6 +198,16 @@ lemma coe_int_dvd [comm_ring α] (m n : ℤ) (h : m ∣ n) :
   (m : α) ∣ (n : α) :=
 ring_hom.map_dvd (int.cast_ring_hom α) h
 
+@[simp] lemma cast_fst {β : Type*} [has_zero α] [has_one α] [has_add α] [has_neg α]
+  [has_zero β] [has_one β] [has_add β] [has_neg β]
+  (n : ℤ) : (n : α × β).fst = n :=
+by induction n; simp *
+
+@[simp] lemma cast_snd {β : Type*} [has_zero α] [has_one α] [has_add α] [has_neg α]
+  [has_zero β] [has_one β] [has_add β] [has_neg β]
+  (n : ℤ) : (n : α × β).snd = n :=
+by induction n; simp *
```

Comment:
These might make more sense as `prod.fst_nat_cast` and `prod.snd_nat_cast`, since they're about `fst (cast ...)` not `cast (fst ...)`.

Reply:
Same for `zmod` as well?